contracts/randdrop/src/state.rs

```diff
@@ -0,0 +1,50 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Addr, Coin, HexBinary, Timestamp, Uint128};
+use cw_storage_plus::{Item, Map};
+
+#[cw_serde]
+pub struct Config {
+    /// manager that can change the manager , register merkle or withdraw funds
+    pub manager: Addr,
+    // The nois-proxy from which to request the randomness
+    pub nois_proxy: NoisProxy,
+    // denom of the token to randdrop
+    pub randdrop_denom: String,
+    /// MerkleRoot is hex-encoded merkle root.
+    pub merkle_root: HexBinary,
+}
+
+#[cw_serde]
+pub struct NoisProxy {
+    // The price to pay the proxy for randomness
+    pub price: Coin,
+    // The address of the nois-proxy contract deployed onthe same chain as this contract
+    pub address: Addr,
+}
+
+#[cw_serde]
+pub struct ParticipantData {
+    // The randomness beacon received from the proxy
+    pub nois_randomness: Option<[u8; 32]>,
+    // amount provided during proof
+    pub base_randdrop_amount: Uint128,
+    // true if the participant won
+    pub is_winner: Option<bool>,
+    // true if the randdrop is claimed
+    pub has_claimed: bool,
```

Comment:
We don't need this field because we can just look at amount_claimed being None or Some, right?

Reply:
yeah we can do that but this is more straight forward API UX wise no?

Reply:
For UX you can expose th query with the interface you want (i.e. query _is result available yet?_). Then this query you implement internally by using as little state as pissible. Here: `amount_claimed.is_some()`

Reply:
okay fair enough. will do